src/Autonomous.cpp

```diff
@@ -478,68 +514,14 @@ void Autonomous::autonomyIter() {
 	const pose_t plan_target_transformed = poseToDraw(plan_target, pose);
 	rospub::publish(plan_target_transformed, rospub::PointPub::PLAN_TARGET);
 
-	int plan_size = plan.rows();
 	pose_t driveTarget = pose;
 	if (nav_state == NavState::GATE_TRAVERSE || dist(plan_target, pose, 0.0) < 2.0) {
 		// We're probably within planning resolution of the goal,
 		// so using the goal as the drive target makes sense.
 		driveTarget = plan_target;
 	} else {
-		// Roll out the plan until we find a target pose a certain distance in front
-		// of the robot. (This code also handles visualizing the plan.)
-		pose_t plan_pose = plan_base;
-
-		// Send starting plan_pose
-		const pose_t start_plan_pose_transformed = poseToDraw(plan_pose, pose);
-		rospub::publish(start_plan_pose_transformed, rospub::PointPub::PLAN_VIZ);
-
-		bool found_target = false;
-		double accumulated_cost = 0;
-		for (int i = 0; i < plan_size; i++) {
-			action_t action = plan.row(i);
-			plan_pose(2) += action(0);
-			plan_pose(0) += action(1) * cos(plan_pose(2));
-			plan_pose(1) += action(1) * sin(plan_pose(2));
-			transform_t tf_plan_pose = toTransform(plan_pose) * invTransform;
-			// We don't care about collisions on already-executed parts of the plan
-			bool plan_pose_collides = (i >= plan_idx);
-			if (USE_MAP) {
-				plan_pose_collides &= map.hasPointWithin(plan_pose, PLAN_COLLISION_DIST);
-			} else {
-				plan_pose_collides &= collides(tf_plan_pose, lidar_scan, PLAN_COLLISION_DIST);
-			}
-			if (plan_pose_collides) {
-				// We'll replan next timestep
-				accumulated_cost += INFINITE_COST;
-				// TODO we should have a more sophisticated way of deciding whether a collision
-				// is imminent.
-				if (dist(plan_pose, pose, 0.0) < PLAN_COLLISION_STOP_DIST) {
-					log(LOG_WARN, "Collision imminent! Stopping the rover!\n");
-					driveTarget = pose; // Don't move
-				}
-			}
-			if (i >= plan_idx && !found_target && dist(plan_pose, pose, 1.0) > FOLLOW_DIST) {
-				found_target = true;
-				plan_idx = i;
-				accumulated_cost += planCostFromIndex(plan, i);
-				driveTarget = plan_pose;
-			} else {
-				// Send current plan_pose to visualization
-				const pose_t curr_plan_transformed = poseToDraw(plan_pose, pose);
-				rospub::publish(curr_plan_transformed, rospub::PointPub::PLAN_VIZ);
-			}
-		}
-		if (!found_target) {
-			plan_idx = plan_size - 1;
-			driveTarget = plan_pose;
-		}
-
-		// This deals with drift
-		// TODO maybe we should just replan if the distance drifts too far
-		accumulated_cost += 2 * dist(plan_pose, plan_target, 1.0);
-		plan_cost = accumulated_cost;
+		driveTarget = getDriveTargetFromPlan(pose, plan_target, lidar_scan);
```

Comment:
Sorry this part looks so confusing---I pulled out the code into this `else` block into a separate function. The code right here is not actually new, just moved.